lib/AckGroupingTracker.cc

```diff
@@ -42,6 +44,14 @@ void AckGroupingTracker::doImmediateAck(const MessageId& msgId, ResultCallback c
         }
         return;
     }
+    if (auto chunkMessageId =
+            std::dynamic_pointer_cast<ChunkMessageIdImpl>(Commands::getMessageIdImpl(msgId))) {
+        auto msgIdList = chunkMessageId->moveChunkedMessageIds();
+        doImmediateAck(std::set<MessageId>(std::make_move_iterator(msgIdList.begin()),
+                                           std::make_move_iterator(msgIdList.end())),
```

Comment:
If the `ackType` is cumulative, we should acknowledge the last message id cumulatively instead of acknowledging all message ids individually.

Reply:
+1